### PR TITLE
Use the isGatheringData selector to differentiate between zero and lo…

### PR DESCRIPTION
**Pull request by asvinb**
…ading states.

## Summary

Addresses issue #4270 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/adminbar/AdminBarClicks.js [hunk 1]
```diff
@@ -73,17 +76,17 @@ function AdminBarClicks( { WidgetReportZero, WidgetReportError } ) {
...
 	if ( error ) {
 		return (
 			<WidgetReportError moduleSlug="search-console" error={ error } />
 		);
 	}
 
-	if ( isZeroReport( searchConsoleData ) ) {
+	if ( ! hasFinishedResolution || isGatheringData === undefined ) {
+		return <PreviewBlock width="auto" height="59px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/adminbar/AdminBarClicks.js:87]: Can you clarify why you moved this check overall? Why can't it remain above the `error` check like it has been so far?

File: assets/js/components/adminbar/AdminBarImpressions.js [hunk 2]
```diff
@@ -73,17 +76,17 @@ function AdminBarImpressions( { WidgetReportZero, WidgetReportError } ) {
...
 	if ( error ) {
 		return (
 			<WidgetReportError moduleSlug="search-console" error={ error } />
 		);
 	}
 
-	if ( isZeroReport( searchConsoleData ) ) {
+	if ( ! hasFinishedResolution || isGatheringData === undefined ) {
+		return <PreviewBlock width="auto" height="59px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/adminbar/AdminBarImpressions.js:87]: See above, why was the overall if clause moved here?

File: assets/js/components/adminbar/AdminBarUniqueVisitors.js [hunk 3]
```diff
@@ -72,15 +75,15 @@ const AdminBarUniqueVisitors = ( { WidgetReportZero, WidgetReportError } ) => {
...
-	}
-
 	if ( error ) {
 		return <WidgetReportError moduleSlug="analytics" error={ error } />;
 	}
 
-	if ( isZeroReport( analyticsData ) ) {
+	if ( ! hasFinishedResolution || isGatheringData === undefined ) {
+		return <PreviewBlock width="auto" height="59px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/adminbar/AdminBarUniqueVisitors.js:84]: Same as above.

File: assets/js/components/wp-dashboard/WPDashboardClicks.js [hunk 4]
```diff
@@ -77,17 +80,17 @@ const WPDashboardClicks = ( { WidgetReportZero, WidgetReportError } ) => {
...
 	if ( error ) {
 		return (
 			<WidgetReportError moduleSlug="search-console" error={ error } />
 		);
 	}
 
-	if ( isZeroReport( data ) ) {
+	if ( loading || isGatheringData === undefined ) {
+		return <PreviewBlock width="48%" height="92px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/wp-dashboard/WPDashboardClicks.js:91]: Same as above.

File: assets/js/components/wp-dashboard/WPDashboardImpressions.js [hunk 5]
```diff
@@ -77,17 +80,17 @@ const WPDashboardImpressions = ( { WidgetReportZero, WidgetReportError } ) => {
...
 	if ( error ) {
 		return (
 			<WidgetReportError moduleSlug="search-console" error={ error } />
 		);
 	}
 
-	if ( isZeroReport( data ) ) {
+	if ( loading || isGatheringData === undefined ) {
+		return <PreviewBlock width="48%" height="92px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/wp-dashboard/WPDashboardImpressions.js:91]: Same as above.

File: assets/js/components/wp-dashboard/WPDashboardPopularPages.js [hunk 6]
```diff
@@ -96,14 +99,18 @@ export default function WPDashboardPopularPages( props ) {
...
+		return null;
 	}
 
 	if ( error ) {
 		return <WidgetReportError moduleSlug="analytics" error={ error } />;
 	}
 
+	if ( loading ) {
+		return <PreviewTable rows={ 6 } />;
+	}
```
Comment (felixarntz) [inline assets/js/components/wp-dashboard/WPDashboardPopularPages.js:112]: Same as above.

File: assets/js/components/wp-dashboard/WPDashboardPopularPages.js [hunk 7]
```diff
@@ -96,14 +99,18 @@ export default function WPDashboardPopularPages( props ) {
 		return data;
 	} );
 
-	if ( loading ) {
-		return <PreviewTable rows={ 6 } />;
+	if ( ! isGatheringData ) {
+		return null;
 	}
```
Comment (felixarntz) [inline assets/js/components/wp-dashboard/WPDashboardPopularPages.js:104]: This isn't really accurate - `isGatheringData` could be `undefined`, in which case this should be in a loading state I think?

File: assets/js/components/wp-dashboard/WPDashboardSessionDuration.js [hunk 8]
```diff
@@ -74,15 +77,15 @@ const WPDashboardSessionDuration = ( {
...
-	}
-
 	if ( error ) {
 		return <WidgetReportError moduleSlug="analytics" error={ error } />;
 	}
 
-	if ( isZeroReport( data ) ) {
+	if ( loading || isGatheringData === undefined ) {
+		return <PreviewBlock width="48%" height="92px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/wp-dashboard/WPDashboardSessionDuration.js:86]: Same as above.

File: assets/js/components/wp-dashboard/WPDashboardUniqueVisitors.js [hunk 9]
```diff
@@ -72,15 +75,15 @@ const WPDashboardUniqueVisitors = ( {
...
-	}
-
 	if ( error ) {
 		return <WidgetReportError moduleSlug="analytics" error={ error } />;
 	}
 
-	if ( isZeroReport( data ) ) {
+	if ( loading || isGatheringData === undefined ) {
+		return <PreviewBlock width="48%" height="92px" />;
+	}
```
Comment (felixarntz) [inline assets/js/components/wp-dashboard/WPDashboardUniqueVisitors.js:84]: Same as above.

**Review by felixarntz — COMMENTED** [review]
@asvinb Mostly looks good, just one primary thing throughout the PR I have a question on.

File: assets/js/components/adminbar/AdminBarClicks.js (hunk 1 above)
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarClicks.js]: @felixarntz I think i know what happened. Basically I searched for `isGatheringData` in the codebase and I came across a file which had the `error` component rendered before the loading one. I thought I had to move it also but turns out it's not the same. So I reverted all the changes I made where I moved the component rendering the error.

File: assets/js/components/adminbar/AdminBarImpressions.js (hunk 2 above)
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarImpressions.js]: See my comment here: https://github.com/google/site-kit-wp/pull/4326#discussion_r745451983

File: assets/js/components/wp-dashboard/WPDashboardClicks.js (hunk 4 above)
Comment (asvinb) [reply assets/js/components/wp-dashboard/WPDashboardClicks.js]: See my comment here: https://github.com/google/site-kit-wp/pull/4326#discussion_r745451983

File: assets/js/components/wp-dashboard/WPDashboardImpressions.js (hunk 5 above)
Comment (asvinb) [reply assets/js/components/wp-dashboard/WPDashboardImpressions.js]: See my comment here: https://github.com/google/site-kit-wp/pull/4326#discussion_r745451983

File: assets/js/components/wp-dashboard/WPDashboardPopularPages.js (hunk 6 above)
Comment (asvinb) [reply assets/js/components/wp-dashboard/WPDashboardPopularPages.js]: See my comment here: https://github.com/google/site-kit-wp/pull/4326#discussion_r745451983

File: assets/js/components/wp-dashboard/WPDashboardPopularPages.js (hunk 7 above)
Comment (asvinb) [reply assets/js/components/wp-dashboard/WPDashboardPopularPages.js]: 👍

File: assets/js/components/adminbar/AdminBarUniqueVisitors.js (hunk 3 above)
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarUniqueVisitors.js:80]: @felixarntz For this one, the error rendering logic stayed where it was. Do you think we should move it above the loading logic to be consistent with how the other widgets are?

File: assets/js/components/wp-dashboard/WPDashboardSessionDuration.js (hunk 8 above)
Comment (asvinb) [reply assets/js/components/wp-dashboard/WPDashboardSessionDuration.js:82]: See my comment here: https://github.com/google/site-kit-wp/pull/4326/files#r745453419

File: assets/js/components/wp-dashboard/WPDashboardUniqueVisitors.js (hunk 9 above)
Comment (asvinb) [reply assets/js/components/wp-dashboard/WPDashboardUniqueVisitors.js:80]: See my comment here: https://github.com/google/site-kit-wp/pull/4326/files#r745453419

File: assets/js/components/adminbar/AdminBarUniqueVisitors.js (hunk 3 above)
Comment (felixarntz) [reply assets/js/components/adminbar/AdminBarUniqueVisitors.js:80]: I think this is looking good as is.

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**